File: FootballExplorer_java/app/src/main/java/fr/a42consulting/footballexplorer/Presenter/TeamsSearchPresenter.java
package fr.a42consulting.footballexplorer.Presenter;

import org.json.JSONException;
import org.json.JSONObject;

import fr.a42consulting.footballexplorer.HttpGetRequest;
import fr.a42consulting.footballexplorer.Model.TeamsSearchModel;
import fr.a42consulting.footballexplorer.View.ITeamsSearchView;

public class TeamsSearchPresenter implements ITeamsSearchPresenter {
    private ITeamsSearchView searchView;
    private TeamsSearchModel league;

    public TeamsSearchPresenter(ITeamsSearchView searchView) {
        this.searchView = searchView;
    }

    @Override
    public void onSearch(String request) {
        // effectuer la recherche ici

        HttpGetRequest getRequest = new HttpGetRequest() {
            @Override
            public void onResponseReceived(String result) {
                if (result == null) {
                    league = new TeamsSearchModel();
                    searchView.onSearchResult(league.getTeams());
                    return;
                }
                try {
                    JSONObject jsonResult = new JSONObject(result);
                    league = TeamsSearchModel.fromJson(jsonResult);
                    searchView.onSearchResult(league.getTeams());
                } catch (JSONException e) {
                    searchView.onMessage("La récupération des données a échoué.");
                    e.printStackTrace();
                }
            }
        };

        getRequest.execute("https://www.thesportsdb.com/api/v1/json/1/search_all_teams.php", "l", request);

    }

    @Override
    public void onSelectTeam(int position) {
        String teamName = league.getTeams().get(position).getName();
        searchView.onShowTeam(teamName);
    }
}

File: FootballExplorer_kotlin/app/src/main/java/fr/a42consulting/footballexplorer/presenter/TeamsSearchPresenter.kt
package fr.a42consulting.footballexplorer.presenter

import org.json.JSONException
import org.json.JSONObject

import fr.a42consulting.footballexplorer.model.TeamsSearchModel
import fr.a42consulting.footballexplorer.view.ITeamsSearchView
import org.jetbrains.anko.doAsync
import org.jetbrains.anko.uiThread
import java.net.URL
import java.net.URLEncoder

class TeamsSearchPresenter(private var searchView: ITeamsSearchView?) : ITeamsSearchPresenter {
    private var league: TeamsSearchModel? = null

    override fun onSearch(request: String) {
        doAsync {
            val result = URL("https://www.thesportsdb.com/api/v1/json/1/search_all_teams.php?l=" + URLEncoder.encode(request, "UTF-8")).readText()
            uiThread {
                try {
                    val jsonResult = JSONObject(result)
                    league = TeamsSearchModel.fromJson(jsonResult)
                    searchView!!.onSearchResult(league!!.teams, true)
                } catch (e: JSONException) {
                    searchView!!.onMessage("La récupération des données a échoué.")
                    e.printStackTrace()
                }
            }
        }
    }

    override fun onSelectTeam(position: Int) {
        val teamName = league!!.teams[position].name
        searchView!!.onShowTeam(teamName)
    }

    override fun resetView(view: ITeamsSearchView?) {
        searchView = view
        if (league != null && searchView != null) {
            searchView!!.onSearchResult(league!!.teams, false)
        }
    }
}
